build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CreateAndroidEmulator.cs

```diff
@@ -1,55 +1,75 @@
 {
-	public class CreateAndroidEmulator : Task
+	public class CreateAndroidEmulator : ToolTask
 	{
 		public                  string          SdkVersion      {get; set;}
 		public                  string          AndroidAbi      {get; set;}
 		public                  string          AndroidSdkHome  {get; set;}
 		public                  string          JavaSdkHome     {get; set;}
 
-		public                  string          ToolPath        {get; set;}
-		public                  string          ToolExe         {get; set;}
-
 		public                  string          TargetId        {get; set;}
 
 		public                  string          ImageName           {get; set;} = "XamarinAndroidTestRunner64";
+		public                  string          DeviceName          {get; set;} = "pixel_4";
 
 		public                  string          DataPartitionSizeMB {get; set;} = "2048";
 		public                  string          RamSizeMB           {get; set;} = "2048";
 
+		protected override string ToolName => "avdmanager";
 
 		public override bool Execute ()
 		{
 			if (string.IsNullOrEmpty (TargetId) && !string.IsNullOrEmpty (SdkVersion)) {
 				TargetId    = "system-images;android-" + SdkVersion + ";default;" + AndroidAbi;
 			}
-			Log.LogMessage (MessageImportance.Low, $"Task {nameof (CreateAndroidEmulator)}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidAbi)}: {AndroidAbi}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdkHome)}: {AndroidSdkHome}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (JavaSdkHome)}: {JavaSdkHome}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ImageName)}: {ImageName}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (SdkVersion)}: {SdkVersion}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (TargetId)}: {TargetId}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ToolExe)}: {ToolExe}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (ToolPath)}: {ToolPath}");
-
-			Run (GetAndroidPath ());
+
+			base.Execute ();
+
+			if (Log.HasLoggedErrors)
+				return false;
+
+			string configPath = Path.Combine (Environment.GetFolderPath (Environment.SpecialFolder.Personal), ".android", "avd", $"{ImageName}.avd", "config.ini");
+			if (!File.Exists (configPath)) {
+				Log.LogWarning ($"Config file for AVD '{ImageName}' not found at {configPath}");
+				Log.LogWarning ($"AVD '{ImageName}' will use default emulator settings (memory and data partition size)");
+			} else {
+				ulong diskSize;
+				if (!UInt64.TryParse (DataPartitionSizeMB, out diskSize))
+					Log.LogError ($"Invalid data partition size '{DataPartitionSizeMB}' - must be a positive integer value expressing size in megabytes");
+
+				ulong ramSize;
+				if (!UInt64.TryParse (RamSizeMB, out ramSize))
+					Log.LogError ($"Invalid RAM size '{RamSizeMB}' - must be a positive integer value expressing size in megabytes");
+
+				File.AppendAllLines (configPath, new string [] {
```

Comment:
What keeps `configPath` from having these values appended to it every time this task runs?   How does `avdmanager` behave if/when `configPath` contains the "same" values appended to the end of the file?

Reply:
Hmm, yeah it looks like it would always append... I guess it always was like this?

I will look into fixing this tomorrow, can probably just read every line of the file and only append if there isn't a match.

Reply:
I reworked the task, so it only writes the file when needed.